src/main/java/com/ibm/nfvodriver/driver/NSLifecycleManagementDriver.java

```diff
@@ -92,14 +96,16 @@ public NSLifecycleManagementDriver(AuthenticatedRestTemplateService authenticate
      * @return newly created {@link NsInstance} record
      * @throws SOL005ResponseException if there are any errors creating the NS instance
      */
-    public String createNsInstance(final ResourceManagerDeploymentLocation deploymentLocation, final String createNsRequest) throws SOL005ResponseException {
+    public String createNsInstance(final ResourceManagerDeploymentLocation deploymentLocation, final String createNsRequest, final String driverrequestid) throws SOL005ResponseException {
         final String url = deploymentLocation.getProperties().get(NFVO_SERVER_URL) + API_CONTEXT_ROOT + API_PREFIX_NS_INSTANCES;
         final HttpHeaders headers = getHttpHeaders(deploymentLocation);
         headers.setContentType(MediaType.APPLICATION_JSON);
         final HttpEntity<String> requestEntity = new HttpEntity<>(createNsRequest, headers);
+        UUID uuid = UUID.randomUUID();
+        LoggingUtils.logEnabledMDC(createNsRequest, MessageType.REQUEST, MessageDirection.SENT, uuid.toString(),MediaType.APPLICATION_JSON.toString(), "http",null,driverrequestid);
```

Comment:
At least url can be part of the protocol_metadata of http requests. This is applicable for all requests.